communications/models.py

```diff
@@ -1,3 +1,192 @@
+    related_project_id = models.PositiveIntegerField(null=True, blank=True)
+    related_message_id = models.PositiveIntegerField(null=True, blank=True)
+    
+    priority = models.CharField(
+        max_length=10,
+        choices=[
+            ('low', _('Low')),
+            ('medium', _('Medium')),
+            ('high', _('High'))
+        ],
+        default='medium'
+    )
+    is_read = models.BooleanField(default=False)
+    expires_at = models.DateTimeField(null=True, blank=True)
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+
+    class Meta:
+        ordering = ['-created_at']
+        indexes = [
+            models.Index(fields=['user', 'is_read', 'created_at'], name='idx_notif_user_read'),
+            models.Index(fields=['user', 'notification_type', 'created_at'], name='idx_notif_user_type'),
+            models.Index(fields=['triggered_by_user'], name='idx_notif_triggered_by'),
+            models.Index(fields=['related_startup_id'], name='idx_notif_startup'),
+            models.Index(fields=['related_project_id'], name='idx_notif_project'),
+            models.Index(fields=['expires_at'], name='idx_notif_expires'),
+        ]
+
+    def __str__(self):
+        return f"{self.get_notification_type_display()} - {self.user.email}"
+
+
+class UserNotificationPreference(models.Model):
+    """Model to store user notification preferences."""
+    user = models.OneToOneField(
+        settings.AUTH_USER_MODEL,
+        on_delete=models.CASCADE,
+        related_name='notification_preferences',
+        primary_key=True
+    )
+    enable_in_app = models.BooleanField(
+        _('Enable in-app notifications'),
+        default=True
+    )
+    enable_email = models.BooleanField(
+        _('Enable email notifications'),
+        default=True
+    )
+    enable_push = models.BooleanField(
+        _('Enable push notifications'),
+        default=False
+    )
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+
+    def __str__(self):
+        return f"Preferences for {self.user.email}"
+
+
+class UserNotificationTypePreference(models.Model):
+    """Model to store user preferences for specific notification types."""
+    user_preference = models.ForeignKey(
+        UserNotificationPreference,
+        on_delete=models.CASCADE,
+        related_name='type_preferences'
+    )
+    notification_type = models.ForeignKey(
+        NotificationType,
+        on_delete=models.CASCADE,
+        related_name='user_preferences'
+    )
+    frequency = models.CharField(
+        max_length=20,
+        choices=NotificationFrequency.choices,
+        default=NotificationFrequency.IMMEDIATE
+    )
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+
+    class Meta:
+        unique_together = ('user_preference', 'notification_type')
```

Comment:
`unique_together` is now deprecated in favor of `UniqueConstraint` in `Meta.constraints`. Update to use `UniqueConstraint` for forward compatibility.